internal/exec/util/blkid.go

```diff
@@ -53,18 +131,8 @@ func filesystemLookup(device string, fieldName string) (string, error) {
 	cFieldName := C.CString(fieldName)
 	defer C.free(unsafe.Pointer(cFieldName))
 
-	switch C.blkid_lookup(cDevice, cFieldName, (*C.char)(unsafe.Pointer(&buf[0])), C.size_t(len(buf))) {
-	case C.RESULT_OK:
-		// trim off tailing NULLs
-		return string(buf[:bytes.IndexByte(buf[:], 0)]), nil
-	case C.RESULT_OPEN_FAILED:
-		return "", fmt.Errorf("failed to open %q", device)
-	case C.RESULT_PROBE_FAILED:
-		// If the probe failed, there's no filesystem created yet on this device
-		return "", nil
-	case C.RESULT_LOOKUP_FAILED:
-		return "", fmt.Errorf("failed to lookup filesystem type of %q", device)
-	default:
-		return "", fmt.Errorf("unknown error")
+	if err := cResultToErr(C.blkid_lookup(cDevice, cFieldName, (*C.char)(unsafe.Pointer(&buf[0])), C.size_t(len(buf))), device); err != nil {
```

Comment:
That was actually a bug in the original cgo implementation. `blkid_do_probe` returns 0 or 1 on success (1 indicates "done" where 0 means "still more to do"). We previously were just checking if it was 0 for success which always failed since it was returning 1. There is also a change to fix this so we actually handle the error instead of just disregarding that case